cmd/controller/controller.go

```diff
@@ -265,7 +265,7 @@ func isNotFound(err error) bool {
 }
 
 func resolveURL(base, ref string) (string, error) {
-	base = strings.TrimSpace(base)
+	base = strings.TrimSuffix(strings.TrimSpace(base), "index.yaml")
```

Comment:
... Dude, you've just hardcoded a special case for URLs that end in the characters "index.yaml" :confused: Apart from "wat" - see my earlier comment for some of the other cases this function also fails to handle correctly, like `resolveURL("https://foo.com/path/myindex.yaml", "http:/otherpath/foo.tgz")` -> `http://foo.com/otherpath/foo.tgz`

(I don't understand why you haven't followed any of the suggestions for ways to do this correctly?  Is there something wrong with my earlier suggestions?)

Reply:
We want `resolveURL` return the absolute URL to the chart tarball. The version you posted in https://github.com/bitnami-labs/helm-crd/pull/23#discussion_r183299565 does not do that. It returns the base url if it cab be parsed by `url.Parse` and if not same with the ref url. so calling the function with the args 

resolveURL("http://chart.example.com/repo", "wordpress-0.0.1.tgz") would return "http://chart.example.com/repo", no?

Reply:
Aha, I see my comment above was too brief:
> If you don't want to use that version (based around URL.Parse) for some reason (older golang?), I suggest you instead use URL.ResolveReference to do the heavy lifting.

There are confusingly *two different* `Parse()` functions in `net/url`.  One is the package-level `url.Parse()`, which parses an absolute or relative URL.  The other is the object-level `URL.Parse()` which parses an absolute or relative URL *in the context of a base URL*, which is exactly what we want here.  Please look again at https://github.com/bitnami-labs/helm-crd/pull/23#discussion_r183299565  (I see I had a typo there on the critical line, which was almost certainly confusing).

Reply:
are you sure about `resolveURL("https://foo.com/path/myindex.yaml", "http:/otherpath/foo.tgz")` -> `http://foo.com/otherpath/foo.tgz`?? The function would return `http:/otherpath/foo.tgz` in this case

Reply:
okay sorry.. i misunderstood your implementation

Reply:
> are you sure about resolveURL("https://foo.com/path/myindex.yaml", "http:/otherpath/foo.tgz") -> http://foo.com/otherpath/foo.tgz?

Yep sorry, I'd forgotten how scheme-relative URLs worked.  The hairy example I was thinking of was `resolveURL("https://foo.com/path/myindex.yaml", "//other.org/foo/bar.tgz")` -> `https://other.org/foo/bar.tgz`

> i misunderstood your implementation but it requires that the base url is appended with trailing /

Correct, that's how relative URLs are defined.  Note that if you pass a regular full URL (like repoURL) in the first place, you get the correct desired behaviour so it's only your test cases that are "wrong" here.  You'll note many webservers redirect `http://foo.com/bar` to `http://foo.com/bar/` (add the trailing slash) for "directory" URLs for this very reason.